### ISSUE TITLE
Rule proposal: Disallow blank lines after @Decorator

### ISSUE THREAD

**Issue by sgarner** [typescript-eslint/typescript-eslint#646]
Is there currently any rule that can check for blank lines between an `@Decorator` and the thing being decorated? I couldn't find any rules for decorators at all in the docs.

This is a code style no-no in my opinion, so it would be good to have a rule to catch this.

**Repro**

```TS
@Decorator()

class Foo {
}
```

**Expected Result**

Error on line 2: blank line(s) after decorator.

**Actual Result**

Lint passes.

**Comment by bradzacher** [comment]
no - there's no rules for decorator spacing.
They're in a really weird spot right now, semi supported by typescript and not maintained too well right now because their spec is in flux a bit.

Happy to accept a PR for this, but it'd have to support class decorators, property/method decorators and argument decorators (I think that's all of the valid usages of them right now).

**Comment by bradzacher** [comment]
If anything, I'd suggest you use `prettier` to do this. I haven't ever really liked stylistic/layout rules because they're so hard to do right, and they're hard to maintain.

[Example showing prettier doing exactly what you want](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABAInSAnAhjCbACgEoAdKCsAG1wGc6ACAMQgkeAscYy0P0KkujYQFsAngFk4MABakOAX2HDeOAdhVRuAB2wQdjALyMAjAG4KYqTNkAmIsO5r+BTdu4iP3XNgDmpsZmwiRKFIogADQgBjAAltB0yKC++gDuAAq+CEkouABuEHEAJlEgAEZ4YADWMgDKOrhgcVB+yDDYAK5w0bIwotQA6rJx8HSNYHB1OaNx+aPiyOAMZS10cNgwGXh+orjIAGa41OvRAFZ0AB4AQlW1MHW4onAAMi1wh8enIBeXdS1+ahwACKnQg8E+Jx6IEa2HW2CWMHEOjgdDA2DiOhgZT0LRggxKcmQAA4AAzRPQQdaDPA6JZ6VEbfIfaIARzB8G2BlyIHoAFooHA4MVhWVsHB2XFxdtcLt9kgjlDoutRHF2l1oXQAUDQeCPgqvtCYLhygTikSkHZoh1cHFqACAMIQUR7JZQaAskCddYAFRNuUV60UiiAA)

Though unfortunately you can't have just a tiny piece of prettier - it's all or nothing 😅

**Comment by JoshuaKGoldberg** [comment]
Per https://eslint.org/blog/2020/05/changes-to-rules-policies#what-are-the-changes, ESLint core is no longer accepting stylistic rules:

> Stylistic rules are frozen - we won't be adding any more options to stylistic rules. We've learned that there's no way to satisfy everyone's personal preferences, and most of the rules already have a lot of difficult-to-understand options. Stylistic rules are those related to spacing, conventions, and generally anything that does not highlight an error or a better way to do something. Update 2021-01-29: We clarified in the README that we will still support newly-added ECMAScript features.

Thanks for the issue report - let's hope decorators are ratified ... eventually! soon!
